File: app/src/main/java/com/bugrui/livedatabus/MainActivity.java
package com.bugrui.livedatabus;

import android.content.Intent;
import android.os.Bundle;

import com.bugrui.buslib.LiveDataBus;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.lifecycle.Observer;

import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private FloatingActionButton fab;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this,Main2Activity.class));
            }
        });

        LiveDataBus.<String>get("main")
                .observe(this, new Observer<String>() {
                    @Override
                    public void onChanged(String s) {
                        Toast.makeText(MainActivity.this,s,Toast.LENGTH_SHORT).show();
                    }
                });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {

        int id = item.getItemId();

        if (id == R.id.action_settings) {
            LiveDataBus.send("main","来自Main1的消息");
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/bugrui/livedatabus/MainActivity.kt
package com.bugrui.livedatabus

import android.content.Intent
import android.os.Bundle
import android.util.Log

import com.bugrui.buslib.LiveDataBus
import com.google.android.material.floatingactionbutton.FloatingActionButton

import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.Observer

import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import java.sql.DriverManager.println

class MainActivity : AppCompatActivity() {

    private var fab: FloatingActionButton? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        fab = findViewById(R.id.fab)
        fab!!.setOnClickListener {
            startActivity(Intent(this@MainActivity, Main2Activity::class.java))
        }

        LiveDataBus.with(10010)
                .observeForever(this, Observer {
                    Log.e("bugrui","MainActivity收到一条普通事件消息")
                })

    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        val id = item.itemId

        if (id == R.id.action_settings) {
            LiveDataBus.sendStickiness(10086, "10086")
            return true
        }

        return super.onOptionsItemSelected(item)
    }
}
